api/v1beta1/clusterclass_types.go

```diff
@@ -158,6 +160,7 @@ type MachineHealthCheckClass struct {
 	// (a) there are at least 3 unhealthy machines (and)
 	// (b) there are at most 5 unhealthy machines
 	// +optional
+	// +kubebuilder:validation:Pattern=^\[[0-9]+-[0-9]+\]$
```

Comment:
That one was definitely missing. I slightly favor validating via marker as you've add it here, given that it's an easy way to get the exactly same validation as we have on the MHC.

Reply:
I would prefer to put it in the webhook until we have some resolution to #6398 - I'm kind of uncomfortable about having different levels of validation so it's harder to see everywhere. The only other kubebuilder validation tags we have here are `kubebuilder:validation:Schemaless` which I think are necessary on this level.

That said I don't have any really strong opinions

Reply:
I'm fine either way, just not ideal that it would differ between ClusterClass and MHC for the field and it's not immediately clear to me how exactly we implement `// +kubebuilder:validation:Pattern=^\[[0-9]+-[0-9]+\]$` in Go

Reply:
For ease of implementation let's do it here then, it's a good idea to ensure we have the exact same validation for both ways of creating the MHC.

Reply:
I am fine with keeping it here for simplicity but it is weird that the validation is distributed in different places (kubebuilder and webhook) for MHC in ClusterClass and the same validations are all implemented in in kubebuilder for MachineHealthCheck. 

I guess we will have a better idea once 6398 is resolved.